ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -453,6 +462,40 @@ private void cleanupOldState() {
     localCommands.ifPresent(lc -> lc.processLocalCommandFiles(serviceContext));
   }
 
+  public void cleanupOldStateDirectories(final KsqlConfig configWithApplicationServer) {
+    final String stateDir =
+        configWithApplicationServer
+            .getKsqlStreamConfigProps()
+            .get(StreamsConfig.STATE_DIR_CONFIG)
+            .toString();
+
+    final Set<String> stateStoreNames =
+        ksqlEngine.getPersistentQueries()
+            .stream()
+            .map(PersistentQueryMetadata::getQueryId)
```

Comment:
I think we can use application ID here and it should match the state store directories. Will test locally once my mvn builds 🤪